### ISSUE TITLE
[BUG] Harvester plots_refresh_parameter not keeping track of plots

### ISSUE THREAD

**Issue by wallentx** [Chia-Network/chia-blockchain#8262]
The harvester is losing track of my plots during replotting, and my plot count is going down.

I have 5156 plots. All of my drives are full, so I am currently replotting.
I have a script that, upon completion of a new NFT plot, it identifies an OG plot, deletes it, and moves the new NFT plot in its place.
I am currently creating 64 plots every 24h.
Earlier today, my harvester did indeed show that it was farming `5156` plots. I check my status a few moments ago, and it shows that I'm only farming `5129` plots.

![2021-08-29_01-35](https://user-images.githubusercontent.com/8990544/131241137-9b0aa144-d510-49d8-8e6c-fd3b0346de39.png)
(Then some time passes for the farmer to rediscover all my plots)
![2021-08-29_01-36](https://user-images.githubusercontent.com/8990544/131241230-535680eb-f101-464b-bd3b-cbe2bca1bf7f.png)

Ok, so back to `5156` after restarting the farmer. It lost track of 27 plots.

When I first discovered this bug this morning, there was a Δ of 85 plots that the harvester wasn't tracking.

> 2021-08-28T13:54:20.544 harvester chia.harvester.harvester: INFO  ...  Total 5071 plots

`chia stop all -d`
`chia start farmer`

> 2021-08-28T14:49:27.451 harvester chia.harvester.harvester: INFO  ...  Total 5156 plots

What I suspect:
My config.yaml had been with me since before launch, and I decided to compare it against https://github.com/Chia-Network/chia-blockchain/blob/main/chia/util/initial-config.yaml to see if I needed to refresh it any. There were a few things that I was missing.
I had `plot_loading_frequency_seconds: 120`, which isn't anywhere in the current initial-config.yaml, so I swapped that out with: https://github.com/Chia-Network/chia-blockchain/blob/main/chia/util/initial-config.yaml#L128-L131
I think that this may not be discovering new plots for some reason. I'm going to attempt to remove those lines, and go back to `plot_loading_frequency_seconds: 120`. I'll check back in the morning to see if the count is off.

Chia: 1.2.5
Distro: Arch Linux [x86_64]
Kernel: Linux 5.13.12-arch1-1
Processor: 24 x Intel(R) Xeon(R) CPU E5-2697 v2 @ 2.70GHz
Processor: 24 x Intel(R) Xeon(R) CPU E5-2697 v2 @ 2.70GHz
RAM: 64362 MiB

**Comment by bradyjarvis** [comment]
We are having issues as well with the harvesters not counting all the plots since upgrading to 1.2.5.  We will try to use a fresh config file and see if we can get those missing plots back.  Our configs are very old as well.

**Comment by wallentx** [comment]
Changing the config back to `plot_loading_frequency_seconds: 120` did not seem to do anything.

This may be due to how the new manager bans badbit/failbit plots.
My final plot management scripts for plotting, and replotting, move plots into active harvester directories with the `.plot` extension. If the file is still being copied when the manager finds it, this causes the manager to throw:

> harvester chia.plotting.manager   : ERROR    Failed to open file /<path_to_plot>/plot-k32-<plotId>.plot. badbit or failbit after reading size 4 at position 107547188232 Traceback (most recent call last):                                                                            File "/<path_to>/chia-blockchain/chia/plotting/manager.py", line 191, in process_file        prover = DiskProver(str(file_path))                                                       RuntimeError: badbit or failbit after reading size 4 at position 107547188232

See here:
https://github.com/Chia-Network/chia-blockchain/blob/53149f2e57743e203aba1d25a89786edcf1f78ab/chia/plotting/manager.py#L198-#L199

I don't know if this outright bans it, or puts it in a loop to try to read it again in 20m. The behavior I'm seeing makes me think it's the former.

For now, I've simply modified my re-plot script (https://github.com/wallentx/farm-and-ranch-supply-depot/blob/main/bin/final-plot-management/replot/replot#L83) to move the plot into the active harvester directory with "-TMP" appended, and removing that when it's finished being copied.

I'll follow up if this solves it

**Comment by wallentx** [comment]
Yep, that solved it. It seems like the ban on badbit/failbit lasts until a restart of the farmer?

**Comment by Zensae** [comment]
Yes, agreed.  I am replotting as well (from a different machine) and then copying the files over from external drives.  Previously, while the copy was in progress the plots may be flagged as bad, but it would eventually farm them a little bit after they finished copying.  Now, once they get flagged as bad it does not appear to ever attempt to farm them again.  Closing and re-opening caused it to refresh all the plots and the ones that had been flagged as bad disappeared and started farming as normal.  

Quite annoying when copying file all day (and all night) to have to now restart the client every once in a while to make sure they're being farmed.  Wish I had stuck with the older version...

**Comment by jamiefiedler** [comment]
> Yep, that solved it. It seems like the ban on badbit/failbit lasts until a restart of the farmer?

restarting the harvester alone appears to be sufficient.

`chia start harvester -r`

**Comment by danarbraz** [comment]
On Linux, use rsync to copy plots if you don't want to script changing file names. rsync creates a temporary file while transferring and renames it to the final file name after transfer is complete.

**Comment by xdustinface** [comment]
Can someone with the issue try https://github.com/Chia-Network/chia-blockchain/pull/8279?

**Comment by sdbarrish** [comment]
Same issue....After I copy files to my NAS, I just run: 

![Capture](https://user-images.githubusercontent.com/14863898/131499039-c9bb85a6-3b17-4e42-aa9c-2d8d8b38c6de.PNG)

Annoying as heck!!!

**Comment by sdbarrish** [comment]
Still not fixed in v1.2.6

**Comment by wallentx** [comment]
> Still not fixed in v1.2.6

Have you updated your config.yaml to reflect this https://github.com/Chia-Network/chia-blockchain/blob/main/chia/util/initial-config.yaml#L128-L132

**Comment by sdbarrish** [comment]
William,

Thank you.  I posted an update on my issue in GitHub.

I was missing that section in my config.yaml file.

What I discovered is that when I do the following commands to update CHIA on my remote harvester running in a docker container on my NAS, that the config.yaml file doesn’t get updated if new sections or lines are added, deleted, or modified to reflect the code changes in the new releases.

chia stop all -d
git stash
git fetch
git checkout {version}
sh install.sh

restart docker container

To solve my issue to make sure my config.yaml file was compliant with the most recent release and previous releases since initial setup, I deleted the container, deleted the image, reinstalled from scratch, init ca’s, and restarted the harvester container.

By doing that, the lines of code that you highlighted now appear in the config.yaml file. I’m sure anything that changed in the config.yaml file since v1.2.1 (my initial install version) also got updated too.

Again, thank you for the most direct and helpful response that I received on my issue.

Sincerely,
Scott
